core/src/gl/shaderProgram.cpp

```diff
@@ -84,15 +120,20 @@ GLint ShaderProgram::getAttribLocation(const std::string& _attribName) {
 
 }
 
-GLint ShaderProgram::getUniformLocation(const std::string& _uniformName) {
+GLint ShaderProgram::getUniformLocation(const UniformEntries::UniformEntry* _entry) {
```

Comment:
@karimnaaji  Sounds good!

But for the API I think it could be made a bit simpler :) Untested code below:

```
UniformLocation {
private:
   const char* name; // owned by static s_uniformStringAtoms
   int id;

public:

   UniformLocation(string name) {
           
         // important: use char* so that references stay valid when vector reallocates
         static vector<std::unique_ptr<char*>>  s_uniformStringAtoms;

         auto it = s_uniformStringAtoms.find(name);
         if (it = s_uniformStringAtoms.end()) {
            id = s_uniformStringAtoms.size();

            s_uniformStringAtoms.emplace_back(name.c_str());
            name = s_uniformStringAtoms.back().get();
            return;
         }

         id = it - s_uniformStringAtoms.begin();
         name = it->get();
  }

   UniformLocation(const UniformLocation& _other) {
       name = _other.name;
       id = _other.id;
   }


   static UniformLocation normalMatrix("u_normalMatrix");
    ...
}


void ShaderProgram::setUniformf(const UniformLocation& _loc, float _value) {}

```